### PR TITLE
Project 9

### PR DESCRIPTION
**Pull request by pkuntong**

### CODE REVIEW

File: src/utils/helpers.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    loadingText = "Saving..."
+  ) {
+    if (isLoading) {
+      btn.textContent = loadingText;
+    } else {
+      btn.textContent = defaultText;
+    }
+  }
+  
+  export function handleSubmit(request, evt, loadingText = "Saving...") {
```
Comment (gennady-bars) [inline src/utils/helpers.js:14]: Nice `handleSubmit` function

File: src/utils/Api.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error ${res.status}`);
+  }
+
+  request(url, options) {
+    return fetch(url, options).then(this.checkResponse);
```
Comment (gennady-bars) [inline src/utils/Api.js]: **COULD BE IMPROVED**

You can make a universal method `_request` not to duplicate the `ok` checking, `headers` and `baseUrl`


```js
 _request(endpoint, options = {}) {
    const finalOptions = {
      // add the default headers
      headers: this._headers,
      // spread the other options if provided
      ...options,
    };
      // add the endpoint to the base url
    const url = `${this._baseUrl}${endpoint}`;
    return fetch(url, finalOptions).then(this._checkResponse);
  }

  // usage

  getUserInfo() {
    // pass only the endpoint here
    return this._request("/users/me");
  }

  getInitialCards() {
    // pass only the endpoint here
    return this._request("/cards");
  }

  // and so on



```

File: src/pages/index.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "9761a594-3fe1-46bc-8b2b-b1c551adb983",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getAppInfo()
+  .then(([cards, users]) => {
```
Comment (gennady-bars) [inline src/pages/index.js:24]: Good job that you get the whole info with `Promise.all`

File: src/pages/index.js [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+);
+const cardModal = document.querySelector("#card-modal");
+const cardModalForm = cardModal.querySelector("#card-form");
+const cardSubmitButton = cardModal.querySelector(".modal__submit-btn");
+const imageInput = cardModal.querySelector("#card-link-input");
+const titleInput = cardModal.querySelector("#card-caption-input");
+
+// Avatar form elements
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarModalForm = avatarModal.querySelector("#edit-avatar-form");
```
Comment (gennady-bars) [inline src/pages/index.js]: **COULD BE IMPROVED**

We can get any form using their **unique** attribute `name` or `id` from `document.forms`

```js

const profileForm = document.forms["profile-form"];
const cardForm = document.forms["card-form"];

```

You need to add `name` or `id` first in `index.html` to achieve that.

More info here https://developer.mozilla.org/en-US/docs/Web/API/Document/forms

**Review by gennady-bars — COMMENTED** [review]
**↓Expand to see the whole summary↓**

Hello, Pau!

Great job on submitting your project 9! This time you’ve implemented `Api` integration. So you can create your own simple sites from now on! 

**Parts I really liked in your project:**

* Good job that you **close** modals only in blocks `then`
* Good job that you `catch` possible errors at the end of server requests.
* Good job that you return the button text in `finally`
* Nice `handleSubmit` function
* Nice `request` method
* Good job that you get the whole info with `Promise.all`

There are few things that need to be corrected in your project. They're mostly minor issues that are easy to fix.

**Needs correcting:**

* All comments in the code  and the comments below:
* Please, make the `confirmation` modal match the design
* Please, make the image modal match the design


There are some comments in your project that can help you make it even better. Please check the `Could be improved` comments.

**A reminder:** It’s necessary to fix all comments to get the project accepted. And `Could be improved` comments are optional, they will be up to you if you want to implement them.

Your hard work on this project is truly commendable. While there are still some adjustments needed. If you need any assistance with the feedback or understanding certain concepts, our tutors are here to help. Keep moving forward and maintain your dedication — you are doing an amazing job!✨

**Review by gennady-bars — COMMENTED** [review]
Some remarks haven't been corrected:

* Please, make the `confirmation` modal match the design 
* Please, make the image modal match the design (pay attantion to the `close icon` poisition)

**Review by gennady-bars — COMMENTED** [review]
One little ing is left:


* The `close icon` in the image modal should be **outside** the modal. Only modals with a form has it inside the container. The same thing is with the `confirmation` modal https://snipboard.io/5PpA83.jpg 
* Please, make the `image modal` match the design on 320px (mobile). See how it looks in figma https://snipboard.io/VGcDLv.jpg

**Comment by pkuntong** [comment]
i need figma the measurement of app mobile/ image popup. i couldn't find it in my figma. Thanks

**Comment by gennady-bars** [comment]
> i need figma the measurement of app mobile/ image popup. i couldn't find it in my figma. Thanks

The image modal design was in the previous sprints. You should check it there. But my screenshot shows what should be done

File: README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Project 6: Spots
+
```
Comment (gennady-bars) [inline README.md:2]: * Please, make the modals match the design (`mobile` and `desktop`). The design can be found here https://www.figma.com/design/1qCS9RkiKiVquBhpOJqjZ0/Sprint-5-Project%3A-Spots?node-id=0-1&node-type=canvas&t=Y8QUtuwYfAUZYJn2-0 . Pay attention, that there are 2 designs: `mobile` and `desktop`. They have different font sizes, margins, paddings, close icon positions and so on

**Review by gennady-bars — APPROVED** [review]
You’ve done a great job!

Your project has been accepted. 

Good luck!

**Pull request closed without merging**